components/suggestion/suggestion_box.jsx

```diff
@@ -693,24 +701,28 @@ export default class SuggestionBox extends React.Component {
                     onKeyDown={this.handleKeyDown}
                 />
                 {(this.props.openWhenEmpty || this.props.value.length >= this.props.requiredCharacters) && this.state.presentationType === 'text' &&
-                    <SuggestionListComponent
-                        ariaLiveRef={this.suggestionReadOut}
-                        open={this.state.focused}
-                        pretext={this.pretext}
-                        location={listStyle}
-                        renderDividers={renderDividers}
-                        renderNoResults={renderNoResults}
-                        onCompleteWord={this.handleCompleteWord}
-                        preventClose={this.preventSuggestionListClose}
-                        onItemHover={this.setSelection}
-                        cleared={this.state.cleared}
-                        matchedPretext={this.state.matchedPretext}
-                        items={this.state.items}
-                        terms={this.state.terms}
-                        selection={this.state.selection}
-                        components={this.state.components}
-                        wrapperHeight={this.props.wrapperHeight}
-                    />
+                    <div style={{width: this.state.width}}>
+                        <SuggestionListComponent
+                            ariaLiveRef={this.suggestionReadOut}
+                            open={this.state.focused}
+                            pretext={this.pretext}
+                            location={listStyle}
+                            renderDividers={renderDividers}
+                            renderNoResults={renderNoResults}
+                            onCompleteWord={this.handleCompleteWord}
+                            preventClose={this.preventSuggestionListClose}
+                            onItemHover={this.setSelection}
+                            cleared={this.state.cleared}
+                            matchedPretext={this.state.matchedPretext}
+                            items={this.state.items}
+                            terms={this.state.terms}
+                            selection={this.state.selection}
+                            components={this.state.components}
+                            wrapperHeight={this.props.wrapperHeight}
+                            calculateInputRect={this.calculateInputRect}
```

Comment:
rather than `suggestion_box` calculating the react size and returning that, it'd be better design wise if you either pass in the input and let it calculate the bounds or passed the sizes as a prop, rather than a function and let it be reactive rather than invoking a function to get the sizes.
e.g. a component like this https://github.com/souporserious/react-measure could help make this reactive

Reply:
Here, react-measure is using ResizeObserver to achieve their goal, but on our case we are mainly looking for changes in position, not so much in size. In order to check position, it exists IntersectionObserver, but it is only for checking the ration of intersection between two elements, which is not exactly what we need. I found ["solutions"](https://stackoverflow.com/questions/40251082/an-event-or-observer-for-changes-to-getboundingclientrect) to this using IntersectionObserver, but (1/5) they feel too convoluted for our solution.

The only event that seems to gives us this kind of reactiveness is scroll. But since we only want to check the scroll on the modal, it makes more sense to keep the registering and handling of the event on the "modal_suggestion_list", instead of on the "suggestion_box". Therefore, we should pass some way to calculate the elements size.

The other way to do this, is as your first suggestion: to move the calculateInputRect logic to modal_suggestion_list, and simply pass the input reference through the props. That way everything is handled in the same place.